src/credentials-manager/index.js

```diff
@@ -0,0 +1,145 @@
+   * @param {String} credentials.accessToken required - token used for API calls
+   * @param {String} credentials.tokenType required - type of the token, ex - Bearer
+   * @param {Number} credentials.expiresIn optional - `expiresAt` should not be empty if this is. Used to denote when the token will expire from the issued time
+   * @param {String} credentials.expiresAt optional - `expiresIn` should not be empty if this is. Used to denote when the token will expire. Has precendence over `expiresIn`.
+   * @param {String} credentials.refreshToken optional - used to refresh access token
+   * @param {String} credentials.scope optional - represents the scope of the current token
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async saveCredentials(credentials = {}) {
+    const validateKeys = ['idToken', 'accessToken', 'tokenType'];
+    validateKeys.forEach(key => {
+      if (!credentials[key]) {
+        const json = {
+          error: 'a0.credential_manager.invalid_input',
+          error_description: `${key} cannot be empty`,
+          invalid_parameter: key,
+        };
+        throw new CredentialsManagerError({json, status: 0});
+      }
+    });
+    if (
+      (!credentials.expiresIn || !credentials.expiresIn > 0) &&
+      !credentials.expiresAt
+    ) {
+      const json = {
+        error: 'a0.credential_manager.invalid_input',
+        error_description: `expiresIn or expiresAt should be set`,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+    try {
+      await this.ensureCredentialManagerIsInitialized();
+      await this.Auth0Module.saveCredentials(credentials);
+    } catch (e) {
+      const json = {
+        error: 'a0.credential_manager.invalid',
+        error_description: e.message,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+  }
+
+  /**
+   * Gets the credential that has already been saved
+   *
+   * @param {String} scope optional - the scope to request for the access token. If null is passed, the previous scope will be kept.
+   * @param {String} minTtl optional - the minimum time in seconds that the access token should last before expiration.
+   * @param {Object} parameters optional - additional parameters to send in the request to refresh expired credentials.
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async getCredentials(scope, minTtl = 0, parameters = {}) {
+    try {
+      await this.ensureCredentialManagerIsInitialized();
+      const credentials = await this.Auth0Module.getCredentials(
+        scope,
+        minTtl,
+        parameters,
+      );
+      return credentials;
+    } catch (e) {
+      const json = {
+        error: 'a0.credential_manager.invalid',
+        error_description: e.message,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+  }
+
+  /**
+   * Gets the credential that has already been saved
```

Comment:
We're using "credentials" vs "credential" most other places, we should be consistent.

```suggestion
   * Gets the credentials that has already been saved
```